boards/arm/qemu_cortex_m0/Kconfig.defconfig

```diff
@@ -11,17 +11,6 @@ config BOARD
 
 if SYS_CLOCK_EXISTS
 
-config NRF_TIMER_TIMER
-	bool "nRF Timer Counter (NRF_TIMER0) Timer"
-	depends on CLOCK_CONTROL
-	depends on SOC_COMPATIBLE_NRF
-	select TICKLESS_CAPABLE
-	default y
-	help
-	  This module implements a kernel device driver for the nRF Timer
-	  Counter NRF_TIMER0 and provides the standard "system clock driver"
-	  interfaces.
-
 config NRF_RTC_TIMER
```

Comment:
Do we really need to have the default n definition below?
Obviously, the board needs to have NRF_TIMER_TIMER defined, instread of NRF_RTC_TIMER.
Naturally, it cannot have both.

What is the best solution? If we leave it like this, then NRF_TIMER_TIMER will be defined as default y in Kconfig , while NRF_RTC_TIMER will stay here as default n, and this is confusing

Reply:
Since `bool`s implicitly default to `n`, it could be removed here. You still got the problem that `NRF_RTC_TIMER` will show up in the menuconfig if it has satisfied dependencies though, so someone could go in and enable it (it's helpful to think of Kconfig in terms of how the menuconfig behaves as a sanity check).

Could change it to a choice instead maybe, and put it in `drivers/timers/Kconfig`, or do something with helper symbols (`HAS_*_TIMER` or the like).

Think that's a larger change though. This PR by itself should be a no-op, and cleans things up a bit.

Reply:
>This PR by itself should be a no-op, and cleans things up a bit.

I get that. But it makes the configuration hard to follow. Take a look at the stuff right below:

```
config NRF_RTC_TIMER
     default n

config SYS_CLOCK_HW_CYCLES_PER_SEC
	default 1000000
```

These are all coupled with the timer selection (`config NRF_TIMER_TIMER` option), which you moved to the other file. So, for example, it is harder, now, to understand why we have 1000000 HW cycles / sec :)

Reply:
Should they depend on the timer selection too? At the moment, the user can go in and turn off `NRF_TIMER_TIMER`, and it won't affect these symbols.

A nice thing about moving it is that it shows up in the Board Options menu. `NRF_TIMER_TIMER` is the only symbol that was defined with a prompt in a Kconfig.defconfig file.

Reply:
Yes, at least SYS_CLOCK_HW_CYCLES_PER_SEC depends on the timer selection. So it is 1 000 000 if the TIMER_TIMER is selected.
All I want if possible is, the NRF_TIMER_TIMER definition to lie close to the SYS_CLOCK_HW_CYCLES_PER_SEC definition, as they are sort of coupled. Can they all be moved to Kconfig?